ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionsReader.scala

```diff
@@ -105,31 +106,38 @@ private[dao] final class TransactionsReader(
   def lookupFlatTransactionById(
       transactionId: TransactionId,
       requestingParties: Set[Party],
-  ): Future[Option[GetFlatTransactionResponse]] = {
-    val query =
-      EventsTable.prepareLookupFlatTransactionById(sqlFunctions)(transactionId, requestingParties)
-    dispatcher
-      .executeSql(
-        databaseMetrics = dbMetrics.lookupFlatTransactionById,
-        extraLog = Some(s"tx: $transactionId, parties = ${requestingParties.mkString(", ")}"),
-      ) { implicit connection =>
-        query.asVectorOf(EventsTable.rawFlatEventParser)
+  )(implicit loggingContext: LoggingContext): Future[Option[GetFlatTransactionResponse]] =
+    withEnrichedLoggingContext(
+      "transactionId" -> transactionId,
+      "parties" -> requestingParties.mkString(", "),
```

Comment:
I'd love it if we could have sequences in the logging without using `mkString`, that got serialized to a JSON array.

Reply:
I would too and I definitely do not like having this kind of bespoke formatting embedded in how we use structured logging (see my comments about principles to follow when using structured logging), but I think we'd need to have some serious design discussion about it. I believe there's a lot of value in having complex values in the structured logging framework (i.e. arrays and objects). E.g. we could dump the entire update received by the indexer and basically build https://github.com/digital-asset/daml/issues/6609 _on top of logging_, which in my opinion is a huge win. If we do that, it's very important that we make sure that the design fits well with the Logstash Logback Encoder. I'll open a ticket to make sure we can have a discussion at a later point.

Reply:
Happy for this to be captured in an issue and handled later.

Reply:
Created https://github.com/digital-asset/daml/issues/6945